CannibalisticZombies/Assets/Scripts/InteractionObject.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+
+
+namespace CannibalisticZombies
+{
+  
+    //-/////////////////////////////////////////////////////////////////////
+    ///
+    /// Author : Meghana Indukuri (meghana.indukuri@sjsu.edu)
+    /// 
+    /// This InteractionObject helps with UI and interaction features for each object in the game
+    /// The player will interact based on this input (E)
+    /// 
+    public class InteractionObject : MonoBehaviour, IPointerEnterHandler, IPointerExitHandler
+    {
+        ///  This is a variable that is used to grab the canvas associated with the keyBind UI
+        [SerializeField] public GameObject keyBindUI;
```

Comment:
From what I have learned, using `[SerializeField]` and `public` is redundant since both expose the variable into the inspector. Since we won't need to access keyBindUI from other classes tho, I think it is okay to declare it as private instead and keep `[SerializeField]`